src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -646,7 +646,12 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
-        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
what about using the ternary op?
`@chmod($tmpFile, !file_exists($filename) ? 0666 & ~umask() : fileperms($filename))`